examples/ftd_api_scripts/pppoe_dynamic.py

```diff
@@ -0,0 +1,82 @@
+'''
+Copyright (c) 2019 Cisco and/or its affiliates.
+This software is licensed to you under the terms of the Cisco Sample
+Code License, Version 1.1 (the "License"). A copy of the License
+can be found in the LICENSE.TXT file of this software or at
+https://developer.cisco.com/site/license/cisco-sample-code-license/
+All use of the material herein must be in accordance with the terms of
+the License. All rights not expressly granted by the License are
+reserved. Unless required by applicable law or agreed to separately in
+writing, software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+express or implied.
+'''
+
+from ftd_api_resources.access_token import get_access_token
+from ftd_api_resources.interface import get_intf_object
+from ftd_api_resources.pppoe import update_intf_to_pppoe
+
+
+def pppoe_dynamic(host, port, user, passwd, intf_hw_name='GigabitEthernet1/1', vpdn_grp_name='my_pppoe_group',
+                  pppoe_user='my_pppoe_username', pppoe_password='my_pppoe_password', ppp_auth_type='PAP',
+                  pppoe_metric=1, obtain_default_route=True):
+    """
+    End to end example of code that updates a physical interface to enable PPPoE on it.
+    Requires Python v3.0 or greater and the reqeusts library.
+    You must update the values for host, port, user, and password to connect to your device.
```

Comment:
please include params in docstrings. This code is meant to be consumed by non-cisco users, or at least non-fdm developers.